doc/user/layouts/partials/create-source/connector/s3/with-options.html

```diff
@@ -0,0 +1,24 @@
+`region` | `text` | **required** A valid AWS region.
+
+#### AWS Credentials `WITH` options
+
+Field | Value | Description
+------|-------|------------
+{{ partial (printf "aws-credentials-with-options") . -}}
+
+##### Permissions Required
+
+The IAM User or Role that `materialized` is operating as requires permission to
+perform different AWS actions depending on what **OBJECTS FROM** specifies as a
+key name source.
```

Comment:
Not entirely sure what a "key name source" is and it doesn't seem to be defined anywhere.

Reply:
yeah this is complex. I've rewritten this to use an example and a table to hopefully make it more clear, but I'm not sure if just spamming words at explanation is the most helpful thing I can do.